ctapipe/conftest.py

```diff
@@ -135,3 +136,117 @@ def prod5_proton_simtel_path():
     return get_dataset_path(
         "proton_20deg_0deg_run4___cta-prod5-paranal_desert-2147m-Paranal-dark-100evts.simtel.zst"
     )
+
+
+@pytest.fixture(scope="session")
+def dl1_tmp_path(tmp_path_factory):
+    return tmp_path_factory.mktemp("dl1")
+
+
+@pytest.fixture(scope="session")
+def dl1_file(dl1_tmp_path):
+    """
+    DL1 file containing both images and parameters from a gamma simulation set.
+    """
+    from ctapipe.tools.stage1 import Stage1Tool
+    from ctapipe.core import run_tool
+
+    output = dl1_tmp_path / "images.dl1.h5"
+
+    # prevent running stage1 multiple times in case of parallel tests
+    with FileLock(output.with_suffix(output.suffix + ".lock")):
+        if output.is_file():
+            return output
+
+        infile = get_dataset_path("gamma_test_large.simtel.gz")
+
+        argv = [
+            f"--input={infile}",
+            f"--output={output}",
+            "--write-images",
+            "--max-events=20",
+            "--allowed-tels=[1,2,3]",
+        ]
+        assert run_tool(Stage1Tool(), argv=argv, cwd=dl1_tmp_path) == 0
```

Comment:
Does this create the output file *and* check if it ran correctly? I am recently reading up on coding best practices and this seems like "hidden side effects" (although in this case this side effect is wanted) (I read this line as assert no error, side effect create the file). Maybe split this up into two lines? What are your thoughts on this?

Reply:
This is not "hidden side effects", this is error handling with return codes. Which is not really pythonic, but this here is about the semantics of running command line tools, not really about python.